### ISSUE TITLE
Additional type for documentation of examples?

### ISSUE THREAD

**Issue by asmagill** [Hammerspoon/hammerspoon#1577]
@cmsj, objections to adding the "Sample" type to the documentation? (I think that's the label recognized by Dash, but I'll check before making any changes)

My idea is a section for 10-20 line examples of some of the more complex things you can do with some of the modules -- a little long for sticking into a single method's notes (especially since it will usually involve multiple methods), but still relatively short; anything longer should be a spoon or in the wiki or a link if someone has them up somewhere else already (gist, github repository, blog, etc.)

**Comment by latenitefilms** [comment]
For what it's worth, I'd LOVE to have full code examples for each extension in the API documentation. For example, rather than have to experiment with `hs.streamdeck` it would have been cool to just be able to copy and paste a full example into a test `init.lua` to see how it's supposed to be all slotted together. I added examples to the `hs.dialog` documentation for this very reason.

**Comment by cmsj** [comment]
@asmagill yep, seems like a good idea. We might need to do a few tweaks to build_docs.py, but that's fine.
